pyvista/plotting/widgets.py

```diff
@@ -140,10 +138,11 @@ def _the_callback(box_widget, event_id):
 
     def clear_box_widgets(self):
         """Disable all of the box widgets."""
-        if hasattr(self, 'box_widgets'):
-            for widget in self.box_widgets:
-                widget.Off()
-            del self.box_widgets
+        # this is necessary for garbage collection
+        for widget in list(self.box_widgets):
+            self.box_widgets.remove(widget)
+
+        self.box_widgets = []
```

Comment:
There is also a subtlety here. This might actually be related to the garbage collection bug.

As it is now, `box_widgets` and `_camera_widgets` are class (rather than instance) attributes. When you do `self.box_widgets.append(foo)` you are mutating `WidgetHelper.box_widgets`. All `WidgetHelper`s share this list, because it's tied to the type.

But when you do
```py
self.box_widgets = []
```
you create an _instance_ attribute that shadows the class attribute. From that point on this `WidgetHelper` (this `Plotter`) will have a dedicated set of box widgets.

So two points here:
  1. If we go with my comment that directly clears `self.box_widgets` we circumvent this problem by always mutating the class attribute.
  2. Are these two class attributes on purpose? Shouldn't these widgets be handled on a per-plotter level? This would also mean that a `Plotter`'s demise involves clearing these lists.

Reply:
> There is also a subtlety here. This might actually be related to the garbage collection bug.*

That must be it! Let me fix that.